lib/valueflow.cpp

```diff
@@ -7036,8 +7036,9 @@ static std::vector<ValueFlow::Value> getInitListSize(const Token* tok,
                                                      bool known = true)
 {
     std::vector<const Token*> args = getArguments(tok);
-    // Strings don't use an init list
     if (!args.empty() && container->stdStringLike) {
+        if (args[0]->tokType() == Token::Type::eChar) // init list of chars
```

Comment:
I think it would be better to check with `valueType` if its a char: `args[0]->valueType() && (args[0]->valueType()->type == ValueType::Type::CHAR` || args[0]->valueType()->type == ValueType::Type::WCHAR_T)` as this will just check for literal chars. You could add a `astIsChar` function to astutils.cpp to check this similar to `astIsIntegral` if you wanted to.

Reply:
Thanks for the suggestions.